pkg/landscaper/deployitem/deployitem.go

```diff
@@ -0,0 +1,33 @@
+// SPDX-FileCopyrightText: 2020 SAP SE or an SAP affiliate company and Gardener contributors.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package deployitem
+
+import (
+	"context"
+	"time"
+
+	"github.com/go-logr/logr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	lsv1alpha1 "github.com/gardener/landscaper/apis/core/v1alpha1"
+	lsv1alpha1helper "github.com/gardener/landscaper/apis/core/v1alpha1/helper"
+)
+
+const (
+	DefaultPickupTimeout   = 5 * time.Minute    // default pickup timeout
+	PickupTimeoutReason    = "PickupTimeout"    // for error messages
+	PickupTimeoutOperation = "WaitingForPickup" // for error messages
+)
+
+// FailDueToTimeout sets the phase of the deployitem to failed and sets lastError to a timeout error
+func FailDueToTimeout(ctx context.Context, c client.Client, log logr.Logger, di *lsv1alpha1.DeployItem, timeoutMessage string) error {
```

Comment:
I think for simplicity this function can be directly inline in the reconicle function.
so no need for another package